finetuner/tuner/dataset/__init__.py

```diff
@@ -1,8 +1,184 @@
+import abc
 import itertools
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import numpy as np
 
 from ... import __default_tag_key__
+from ...helper import DocumentSequence
+
+
+class BaseDataset:
+    @abc.abstractmethod
+    def __init__(
+        self,
+        docs: DocumentSequence,
+        preprocess: Optional[Callable[[Union[str, np.ndarray]], Any]] = None,
+    ):
+        ...
+
+    @abc.abstractmethod
+    def __getitem__(
+        self, ind: int
+    ) -> Tuple[Union[str, np.ndarray], Union[int, Tuple[int, int]]]:
+        ...
+
+    @property
+    @abc.abstractmethod
+    def labels(self) -> Union[int, Tuple[int, int]]:
+        ...
```

Comment:
what is this expect label?

Reply:
Depends on dataset - see this method for class and sessions dataset